analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,272 @@
+    ) %>%
+    # Define dominant status based the weighted means meeting a status
+    # threshold
+    mutate(
+      dominant_arm_status = case_when(
+        callable_fraction < (1 - uncallable_threshold) ~ "uncallable",
+        loss_fraction > status_threshold ~ "loss",
+        gain_fraction > status_threshold ~ "gain",
+        loss_fraction + gain_fraction > status_threshold ~ "unstable",
+        TRUE ~ "neutral"
+      )
+    )
+
+# Display table
+consensus_seg_arm_status
+```
+
+### Determine cytoband status
+
+We want to include cytoband and gene-level calls for chromosome arms that have not been defined as a gain or loss to make the cytoband-level majority calls.
+
+```{r}
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than the `status_threshold` fraction
+# value of the total counts are for that particular status
+consensus_seg_cytoband_status <-
+  consensus_seg_cytoband_status_df %>%
+  left_join(
+    consensus_seg_arm_status,
+    by = c(
+      "Kids_First_Biospecimen_ID",
+      "chromosome_arm"
+    )
+  ) %>%
+  mutate(
+    dominant_cytoband_status = case_when(
+      callable_fraction.x < (1 - uncallable_threshold) ~ "uncallable",
+      loss_fraction.x > status_threshold ~ "loss",
+      gain_fraction.x > status_threshold ~ "gain",
+      loss_fraction.x + gain_fraction.x > status_threshold ~ "unstable",
+      TRUE ~ "neutral"
+    )
+  ) %>%
+  # Filter the annotated CN data to include only neutral chromosome arms and disagreements
+  filter(
+    dominant_arm_status %in% c("neutral", "uncallable", "unstable") |
+      (
+        dominant_cytoband_status != dominant_arm_status &
+          # bands that disagree with arm, but are not neutral (or uncallable)
+          !(dominant_cytoband_status %in% c("neutral", "uncallable", "unstable"))
+      )
+  ) %>%
+  select(
+    Kids_First_Biospecimen_ID,
+    cytoband,
+    loss_fraction = loss_fraction.x,
+    gain_fraction = gain_fraction.x,
+    callable_fraction = callable_fraction.x,
+    dominant_cytoband_status
+  )
+
+# Display table
+consensus_seg_cytoband_status
+```
+
+### Determine gene-level status
+
+```{r}
+# Now create a data.frame with the gene-level status calls for the
+# neutral, uncallable, and unstable cytoband-level calls
+consensus_seg_gene_status <- consensus_seg_autosomes_df %>%
+  # Create a cytoband column that keeps only the first half of the ranges
+  # for joining purposes
+  mutate(cytoband_no_ranges = gsub("-.*", "\\1", cytoband)) %>%
+  left_join(
+    consensus_seg_cytoband_status_df,
+    by = c(
+      "biospecimen_id" = "Kids_First_Biospecimen_ID",
+      "cytoband_no_ranges" = "cytoband"
+    )
+  ) %>%
+  # Filter the annotated CN data to include only neutral cytobands and disagreements
+  filter(
+    dominant_cytoband_status %in% c("neutral", "uncallable", "unstable") |
+      (status != dominant_cytoband_status &
```

Comment:
This will catch all of the places where the gene level call is "amplification" because we don't make that call at the cytoband level. I think the right thing to do is to change all "amplification" calls to "gain" for the purposes of this script and its output.  

I also have some thought that we should also be looking at arm-level calls at this point. I can imagine a case where an arm is a **gain** and a cytoband within it is **neutral** and then a gene within that band is a **gain**. We would not be including the cytoband in the final set (on the assumption it is a false negative), but by the current logic we would include the individual gene, but I don't think we should. The logic here may get a bit messy, but I think start is to be sure to join in the arm level calls as well, so we have both `dominant_cytoband_status` and `dominant_arm_status` to look at before making decisions.

Reply:
In the most recent commit, I recoded the "amplification" calls to be "gain" calls as you suggested here.

I also _believe_ that I implemented the logic of including individual gene calls using both the `dominant_cytoband_status` and `dominant_arm_status` data.

Again, please let me know if I misinterpreted this comment in the way I re-formatted the code.